executor/set.go

```diff
@@ -181,8 +183,32 @@ func (e *SetExecutor) setSysVariable(name string, v *expression.VarAssignment) e
 		return stmtsummary.StmtSummaryByDigestMap.SetMaxSQLLength(valStr, !v.IsGlobal)
 	case variable.TiDBCapturePlanBaseline:
 		variable.CapturePlanBaseline.Set(valStrToBoolStr, !v.IsGlobal)
+	case variable.TiDBEnableTopSQL:
+		config.UpdateGlobal(func(conf *config.Config) {
+			conf.TopSQL.Enable = variable.TiDBOptOn(valStr)
+		})
+	case variable.TiDBTopSQLAgentAddress:
+		// todo: add validate
+		config.UpdateGlobal(func(conf *config.Config) {
+			conf.TopSQL.AgentAddress = valStr
+		})
+	case variable.TiDBTopSQLPrecisionSeconds:
+		var val int
+		val, err = strconv.Atoi(valStr)
+		if err == nil {
+			config.UpdateGlobal(func(conf *config.Config) {
+				conf.TopSQL.PrecisionSeconds = val
+			})
+		}
+	case variable.TiDBTopSQLMaxStatementCount:
+		var val int
+		val, err = strconv.Atoi(valStr)
+		if err == nil {
+			config.UpdateGlobal(func(conf *config.Config) {
+				conf.TopSQL.MaxStatementCount = val
+			})
+		}
```

Comment:
Is it possible to do the same here with the setters on the sysvar? (SetSession/SetGlobal). Eventually I would like to remove the special handling of statement summary here and move it to setters.

Reply:
Great, already changed, PTAL.

Reply:
Awesome, thank you!